leaderboard/service/get_member.go

```diff
@@ -0,0 +1,28 @@
+package service
+
+import (
+	"context"
+
+	"github.com/topfreegames/podium/leaderboard/model"
+)
+
+const getMemberServiceLabel = "get member"
+
+// GetMember return a member info
+func (s *Service) GetMember(ctx context.Context, leaderboard, member string, order string, includeTTL bool) (*model.Member, error) {
+	databaseMembers, err := s.Database.GetMembers(ctx, leaderboard, order, includeTTL, member)
+	if err != nil {
+		return nil, NewGeneralError(getMemberServiceLabel, err.Error())
+	}
+
+	if databaseMembers[0] == nil {
+		return nil, NewMemberNotFoundError(leaderboard, member)
+	}
+
+	return &model.Member{
+		PublicID: databaseMembers[0].Member,
+		Score:    int64(databaseMembers[0].Score),
+		Rank:     int(databaseMembers[0].Rank) + 1,
+		ExpireAt: int(databaseMembers[0].TTL),
+	}, nil
+}
```

Comment:
WFYT if we wrapped those parameters in a struct to reduce the number of arguments in those function calls? Something like:

```suggestion
type GetMemberParams struct {
    Leaderboard string
    Member string
    Order string
    IncludeTTL bool
}

// GetMember return a member info
func (s *Service) GetMember(ctx context.Context, params GetMemberParams) (*model.Member, error) {
	databaseMembers, err := s.Database.GetMembers(ctx, params.Leaderboard, params.Order, params.IncludeTTL, params.Member)
	if err != nil {
		return nil, NewGeneralError(getMemberServiceLabel, err.Error())
	}

	if databaseMembers[0] == nil {
		return nil, NewMemberNotFoundError(params.Leaderboard, params.Member)
	}

	return &model.Member{
		PublicID: databaseMembers[0].Member,
		Score:    int64(databaseMembers[0].Score),
		Rank:     int(databaseMembers[0].Rank) + 1,
		ExpireAt: int(databaseMembers[0].TTL),
	}, nil
}
```

Reply:
After some thought, I don't think this is a good idea as it would increase the complexity for game devs to update to the new version. As we want an update as smooth as possible, we will keep the long list of parameters.